internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -50,12 +50,22 @@ var (
 		`"private_key":{"filename":"%s"}}}]}`, XdsTLSCertFilename, XdsTLSKeyFilename)
 )
 
-// ExpectedResourceHashedName returns expected resource hashed name.
+// ExpectedResourceHashedName returns expected resource hashed name including up to the 48 characters of the original name.
 func ExpectedResourceHashedName(name string) string {
-	hashedName := providerutils.GetHashedName(name)
+	hashedName := providerutils.GetHashedName(name, 48)
 	return fmt.Sprintf("%s-%s", config.EnvoyPrefix, hashedName)
 }
 
+// ExpectedContainerPortName returns expected container port name with max length of 15 characters.
+// If mergedGateways is enabled or listener port name is larger than 15 characters it will return partially hashed name up to the 7 characters.
+// Listeners on merged gateways have a infraIR port name {GatewayNamespace}/{GatewayName}/{ListenerName}.
+func ExpectedContainerPortName(infraPortName string) string {
```

Comment:
we are always hashing for `deployment` , `service` and `service Port Name`, why not always hash for `container port name` as well (instead of this if else)

Reply:
I insisted on having this as a listener name, but on the other hand, this is still only 15 characters long.
Let's do a hashed name here as well